.github/workflows/ci.yml

```diff
@@ -8,178 +8,202 @@ on:
+  examples:
+    name: Examples ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: ${{ matrix.shell }}
     env:
-      PKG_CONFIG_ALLOW_CROSS: 1
+      CARGO_BUILD_TARGET: ${{ matrix.target }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Ubuntu
+            os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+            run_examples: true
+            shell: bash
+            setup_env: |
+              set -x;
+              sudo apt update -y
+              sudo apt install -y \
+                cmake ninja-build clang \
+                xorg-dev libwayland-dev libxkbcommon-dev wayland-protocols \
+                libegl1-mesa-dev mesa-vulkan-drivers
+          - name: Macos
+            os: macos-latest
+            target: x86_64-apple-darwin
+            run_examples: false # macos vm doesn't have software metal api
+            shell: bash
+            setup_env: brew install cmake ninja
+          - name: Windows gnu
+            os: windows-latest
+            target: x86_64-pc-windows-gnu
+            run_examples: true
+            shell: msys2 {0}
+          - name: Windows msvc
+            os: windows-latest
+            target: x86_64-pc-windows-msvc
+            run_examples: true
+            shell: cmd
+            setup_env: choco install -y make
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v3
         with:
-          submodules: 'true'
-      - run: rustup component add clippy
-      - name: Install dependencies
-        run: |
-          sudo apt-get update -y -qq
-          sudo apt-get install -y -qq libegl1-mesa-dev libc6-dev-i386
-          echo "$ANDROID_HOME/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin" >> $GITHUB_PATH
-      - run: rustup target add aarch64-linux-android
-      - run: cargo clippy --target aarch64-linux-android
-
-  macos-vulkan-portability:
-    name: MacOS Stable (Vulkan)
-    runs-on: macos-10.15
-    steps:
-      - uses: actions/checkout@v2
+          submodules: true
+      - if: contains(matrix.target, 'windows-msvc')
+        name: Setup msvc
+        uses: ilammy/msvc-dev-cmd@v1
+      - if: contains(matrix.target, 'windows-gnu')
+        name: Setup msys2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: UCRT64
+          update: true
+          install: >-
+            make
+            git
+          pacboy: >-
+            toolchain:p
+            cmake:p
+            ninja:p
+            rust:p
+      - if: "!contains(matrix.target, 'windows-gnu')"
+        name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
         with:
-          submodules: 'true'
-      - run: rustup component add clippy
-      - run: cargo clippy
+          target: ${{ matrix.target }}
+      - name: Setup Environment
+        run: ${{ matrix.setup_env }}
+      - name: Run tests
+        run: cargo test
```

Comment:
So this does not only run the examples. What does `cargo test` do, anyway?

Reply:
it runs some tests that are generated by `bindgen`

https://github.com/gfx-rs/wgpu-native/actions/runs/4687763977/jobs/8307456351#step:7:168